Update and clean up bundled firmware

StollD opened the pull request:
Since we can now install multiple IPTS firmwares next to each other, we can stop having if blocks in the setup script, checking for the model name and extracting firmware conditionally. By putting the firmware under `root/` it gets installed by the same command that installs the config files.

The IPTS firmware files are from the newest release of the driver package for Surface Book 2.

The other big change is that I actually removed any other firmware except the one for IPTS. The reason is that all of these files are part of `linux-firmware`, and get updated through the package manager. The distro most likely can do a better job of keeping them up to date over time than we can.

The only distros that I can think of where these files would be missing are the FSF-approved distros shipping without non-free firmware. If you would want to run such a distro you would have to download and install these files manually. But then, if you would want to run such a distro you probably wouldn't want to buy a Microsoft Surface anyways.

I also pointed the setup script at this repository so that users actually get a kernel that can work with the IPTS layout.

qzed commented:
> The other big change is that I actually removed any other firmware except the one for IPTS. The reason is that all of these files are part of `linux-firmware`, and get updated through the package manager. The distro most likely can do a better job of keeping them up to date over time than we can.

I agree, that already works well for me on Arch and it's definitely more appropriate.



> The only distros that I can think of where these files would be missing are the FSF-approved distros shipping without non-free firmware. If you would want to run such a distro you would have to download and install these files manually. But then, if you would want to run such a distro you probably wouldn't want to buy a Microsoft Surface anyways.

Completely agree. I guess anyone buying a Microsoft device should be aware of that and fine with it.

> I also pointed the setup script at this repository so that users actually get a kernel that can work with the IPTS layout.

Given the absence of Jake and the current development status, that's probably a good idea.

StollD commented:
> I think we should keep the firmware files separate from the config files (something like `firmware/intel/ipts` as separate folder). Apart from that this looks good.

Moved the firmware into their own folder and updated the setup script accordingly.

qzed commented:
Thanks!
